p2p/errors/error.go

```diff
@@ -0,0 +1,18 @@
+package errors
+
+import "fmt"
+
+type ErrFactory func(msg string, err error) error
+
+func NewErrFactory(preMsg string) ErrFactory {
+	return func(msg string, err error) error {
+		msgStr := ""
+		if msg != "" {
+			// NB: with msg - "LibP2P module error: <msg>: <wrapped error>"
+			//  without msg - "LibP2P module error: <wrapped error>"
```

Comment:
```suggestion
			// NB: with msg - "<preMsg>: <msg>: <wrapped error>"
			//  without msg - "<preMsg>: <wrapped error>"
```

Reply:
This is irrelevant, logging conventions are new and the codebase has yet to be made consistent in its usage.